### PR TITLE
Plugin/shutdown

### PR DESCRIPTION
**Pull request by sc0ttr0berts0n**

### CODE REVIEW

File: src/plugins/shutdown/index.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        this.canShutdown = false;
+        this.shutdownPermissionId = null;
+        this.initialMessageID = null;
+    }
+    init() {
+        this.bot.on('message', (message) => {
+            // only check if this is the message *following*
+            // the initial message and can shutdown has been flagged
+            const senderHasPermission =
+                message.member.id === this.shutdownPermissionId;
```
Comment (catcherfreeman) [inline src/plugins/shutdown/index.js:15]: i

File: src/plugins/shutdown/index.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        });
+    }
+    run(message, params = [], command) {
+        console.log(
+            `\n[Shutdown Plugin]\n${message.member.displayName} Initiated Shutdown`
+        );
+
+        message.reply('**Are you sure?** `Y`es /  `N`o');
+        this.initialMessageID = message.id;
+        this.shutdownPermissionId = message.member.id;
```
Comment (catcherfreeman) [inline src/plugins/shutdown/index.js:67]: It looks like you always assign the id of the user that messaged to shutdownPermissionId, then perform the same check on line 15.  I was assuming this would be a fixed list of discord user ids that have permissions (maybe this is a placeholder?)

File: src/plugins/shutdown/index.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                } else {
+                    // otherwise, reinit
+
+                    // log the reversion to console
+                    console.log(
+                        `\n[Shutdown Plugin]\nShutdown Cancelled. Conditions not met.`
+                    );
+
+                    // tell the discord
+                    message.reply('Shutdown Cancelled. Conditions not met.');
```
Comment (catcherfreeman) [inline src/plugins/shutdown/index.js:53]: haha, this seems so robotic, but I guess it's fine

File: src/plugins/shutdown/index.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            if (this.canShutdown && !isOriginalMessage && senderHasPermission) {
+                // clean content
+                const messageContent = message.content.toLowerCase().trim();
+
+                // validity checks
+                const messageisYes =
+                    messageContent === 'y' || messageContent === 'yes';
+
+                // if validity checks are true
+                if (messageisYes && senderHasPermission) {
```
Comment (catcherfreeman) [inline src/plugins/shutdown/index.js]: duplicate check for senderHasPermission here. If we are at this point, it should already be true.

File: src/plugins/shutdown/index.js (hunk 2 above)
Comment (sc0ttr0berts0n) [reply src/plugins/shutdown/index.js:67]: So this *may* be a hack, or may be the right way to go about it. Basically, what's going on here, is that the user who sends the message first (the `initialMessageID` message) gets caught by the event listener. This message is always `!shutdown` so its not `y` or `n` and therefore fails that check. So what I do is a check and see if the message being sent is *not* the initial message, and in fact, a new message from the user with permission to shutdown. 

In short, the bot needs to ignore message one, and wait for message two, from user A. Anything else cancels the sequence.

File: src/plugins/shutdown/index.js (hunk 3 above)
Comment (sc0ttr0berts0n) [reply src/plugins/shutdown/index.js:53]: I could just leave it at "Shutdown Cancelled".

File: src/plugins/shutdown/index.js (hunk 4 above)
Comment (sc0ttr0berts0n) [reply src/plugins/shutdown/index.js]: Good Catch. Shows you took a close look at this.

File: src/plugins/shutdown/index.js (hunk 2 above)
Comment (catcherfreeman) [reply src/plugins/shutdown/index.js:67]: Ahhh okay that makes sense!

File: src/plugins/shutdown/index.js (hunk 3 above)
Comment (catcherfreeman) [reply src/plugins/shutdown/index.js:53]: I like that better :)

Comment (sc0ttr0berts0n) [reply src/plugins/shutdown/index.js:53]: Settled, I'm leaving the logging more "robotic" on the internal log lol.

**Comment by catcherfreeman** [comment]
Looks good
